### PR TITLE
gapic: tighten reference link parser regexp

### PR DESCRIPTION
**Pull request by noahdietz**
I needed to tighten the regexp added in #341. It was capturing too much, incorrectly.

### CODE REVIEW

File: internal/gengapic/markdown.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 )
 
 var linkParser = regexp.MustCompile(`<a href=["'](.+)["']`)
-var referenceParser = regexp.MustCompile(`\[(.+)\]\[(.*)\]`)
+var referenceParser = regexp.MustCompile(`\[([a-zA-Z1-9\.\_]+)\]\[[a-zA-Z1-9\.\_]*\]`)
```
Comment (tbpg) [inline internal/gengapic/markdown.go]: I don't think you need to escape the period or underscore inside a character class (`[...]`).

```suggestion
var referenceParser = regexp.MustCompile(`\[([a-zA-Z1-9._]+)\]\[[a-zA-Z1-9._]*\]`)
```

File: internal/gengapic/markdown_test.go [hunk 2]
```diff
@@ -74,8 +74,8 @@ func TestMDPlain(t *testing.T) {
 			want: "html with a softbreak (at /link/to/some#thing) \n test",
 		},
 		{
-			in:   "link to [a search engine](https://www.google.com) with request type [Search][foo.bar.v1.Search].",
-			want: "link to a search engine (at https://www.google.com) with request type Search.",
+			in:   "link to [a search engine](https://www.google.com) with request type [Search][foo.bar_baz.v1.Search], [biz][].",
```
Comment (tbpg) [inline internal/gengapic/markdown_test.go]: Would you mind adding another test case for something that was matched before this change, but isn't matched now?

File: internal/gengapic/markdown.go (hunk 1 above)
Comment (noahdietz) [reply internal/gengapic/markdown.go]: Done! Thanks, good catch.

File: internal/gengapic/markdown_test.go (hunk 2 above)
Comment (noahdietz) [reply internal/gengapic/markdown_test.go]: Done! I expanded it some more. But basically, if there were multiple references in a block, the previous regex would match the link text to keep as everything between the first square bracket and the closing bracket of the last set, keeping essentially everything.

**Pull request merged**